### PR TITLE
Add support for Trinkets (if present) and add keybinding to open backpack

### PR DESCRIPTION
**Pull request by OmegaJak**
1. Allows placement of any backpack into the backpack trinket slot if the [Trinkets](https://www.curseforge.com/minecraft/mc-mods/trinkets-fabric) mod is present
   - I added a `compileOnly` dependency on Trinkets
   - I used a plugin mixin to conditionally make `BackpackItem` implement `Trinket` when Trinkets is present
   - This adresses the second request in #51 

image

2. Adds a keybinding to open the first backpack in the player's inventory
   - I had to refactor `BackpackScreenHandler` to not be tied to hands for this
   - This addresses the basics for #48, though I didn't go so far as enabling opening of different backpacks by hotkey
   - The search for the 'first' backpack happens in this order:
      1. The backpack slot of the trinket inventory, if present
      2. The player's offhand
      3. Player's hotbar, left to right
      4. Player's main inventory, top left to bottom right

I did my best to match code style & organization to what was already there and your other mods, but change anything as you see fit.

Note: I didn't bother with any rendering stuff while wearing the backpack.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 minecraft_version=1.16.2
 yarn_mappings=1.16.2+build.43
 loader_version=0.9.2+build.206
-fabric_version=0.19.0+build.398-1.16
+fabric_version=0.29.3+1.16
```
Comment (OmegaJak) [inline gradle.properties:8]: Needed this update to use `ClientPlayNetworking` and `ServerPlayNetworking`

**Comment by Draylar** [comment]
Looks awesome, thanks for the fantastic PR! I'll add you to the contributors' list in the fabric.mod.json file before release.

**Comment by Draylar** [comment]
I'm changing the default keybinding to `B` because [Identity](https://www.curseforge.com/minecraft/mc-mods/identity), my morph mod, uses ` for the main menu (and I would prefer for both of them to work together out of the box). Released in 2.1.0 in a second! Thanks again.

**Comment by OmegaJak** [comment]
Sounds good! I had fun doing it

**Pull request merged**